frontend/src/helpers/addBackstageAuthHeaders.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2025 The Kubernetes Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getBackstageToken } from './backstageMessageReceiver';
+
+/**
+ * Adds backstage authentication token to headers if available
+ *
+ * @param headers - optional existing headers to add the token to
+ * @returns headers with backstage token if available
+ */
+export function addBackstageAuthHeaders(headers: HeadersInit = {}): HeadersInit {
+  const token = getBackstageToken();
```

Comment:
Either inside addBackstageAuthHeaders, or at the call site of addBackstageAuthHeaders: it should only do things if it's running isBackstage?

I guess getBackstageToken handles this?

Reply:
Ok, it does. So no backstage stuff is added without backstage being there.